modules/lib/shell.nix

```diff
@@ -41,6 +41,83 @@ rec {
   # statement for each set entry.
   exportAll = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList export vars);
 
+  # Wrap a list of strings to a given line width.
+  # Packs as many items as possible per line without exceeding maxWidth.
+  # Returns a list of strings, each representing a line.
+  #
+  # Example: wrapLines ["item1" "item2" "very-long-item" "item3"] 20
+  #   => ["item1 item2" "very-long-item" "item3"]
+  wrapLines =
+    items: maxWidth:
+    let
+      step =
+        acc: item:
+        let
+          potentialLine = if acc.currentLine == "" then item else "${acc.currentLine} ${item}";
+        in
+        if lib.stringLength potentialLine <= maxWidth then
+          acc // { currentLine = potentialLine; }
+        else
+          acc
+          // {
+            finishedLines = acc.finishedLines ++ [ acc.currentLine ];
+            currentLine = item;
+          };
+      foldResult = lib.foldl' step {
+        finishedLines = [ ];
+        currentLine = "";
+      } items;
+    in
+    foldResult.finishedLines ++ lib.optional (foldResult.currentLine != "") foldResult.currentLine;
+
+  # Format a list of lines into an indented, multi-line block.
+  # Adds 2-space indentation and wraps with newlines for shell array formatting.
+  #
+  # Example: formatMultiLine ["line1" "line2 line3"]
+  #   => "\n  line1\n  line2 line3\n"
+  #
+  # Designed for shell arrays but reusable for any indented multi-line output.
+  formatMultiLine = lines: "\n  ${lib.concatStringsSep "\n  " lines}\n";
+
+  # Formats a list of items for shell array content with intelligent width optimization.
+  # IMPORTANT: This formats the CONTENTS of an array (what goes inside parentheses),
+  # not a complete array definition. Use lib.hm.zsh.define for complete definitions.
+  #
+  # Uses lib.escapeShellArg for robust shell escaping (handles spaces, quotes, special chars).
+  # Packs multiple items per line to optimize terminal width (~78 chars per line).
+  # Short arrays (≤3 items, ≤80 chars total) use single-line format.
+  #
+  # Example outputs:
+  #   Empty:       ""
+  #   Simple:      item1 item2 item3
+  #   With spaces: 'item one' 'item two' 'item three'
+  #   Long arrays: \n  item1 item2 item3\n  item4 item5\n
+  #
+  # Built from composable helpers: quoteShellItems, wrapLines, formatMultiLine
+  formatShellArrayContent =
+    items:
+    # Return early for the empty case
+    if items == [ ] then
+      ""
+    else
+
+      let
+        quotedItems = lib.map lib.escapeShellArg items;
+        singleLine = lib.concatStringsSep " " quotedItems;
+        useSingleLine = (lib.length items <= 3) && (lib.stringLength singleLine <= 80);
+      in
+
+      if useSingleLine then
+        singleLine
```

Comment:
Why the explicit check, rather than relying on `wrapLines` packing the items into the line (which is what it will do, IIUC)?